### ISSUE TITLE
version discrepancies in the 2024-11-03 release

### ISSUE THREAD

**Issue by murphyte** [geneontology/helpdesk#513]
Looking in the 2024-11-03 release files at: https://release.geneontology.org/2024-11-03/ontology/index.html

The go-basic files say the version is 2024-10-27:
go-basic.json:      "version" : "http://purl.obolibrary.org/obo/go/releases/2024-10-27/go.owl"
go-basic.obo:data-version: releases/2024-10-27
go-basic.owl:        <owl:versionIRI rdf:resource="http://purl.obolibrary.org/obo/go/releases/2024-10-27/go.owl"/>

The other files say 2024-11-03:
go.json:      "version" : "http://purl.obolibrary.org/obo/go/releases/2024-11-03/go.owl"
go.obo:data-version: releases/2024-11-03
go.obo:property_value: owl:versionInfo "2024-11-03" xsd:string
go.owl:        <owl:versionIRI rdf:resource="http://purl.obolibrary.org/obo/go/releases/2024-11-03/go.owl"/>
go.owl:        <owl:versionInfo>2024-11-03</owl:versionInfo>

Is the version discrepancy intended? The 2024-10-27 versionIRI URL does not resolve.

**Comment by kltm** [comment]
@murphyte Hm. I can think of a few reasons this might happen. We've seen something similar before here: https://github.com/geneontology/go-ontology/issues/28988 .

Basically, this maybe be a small "dirty workspace" bug, or it may be due to the fact that the pipeline build process works over a few days in some cases.

Also tagging on @balhoff, in case we want to keep this in the go-ontology tracker instead.

May be related to:
https://github.com/geneontology/go-ontology/issues/26372
https://github.com/geneontology/pipeline/issues/162

@balhoff I think I've mentioned it before (I can't find the issue), but it would be good to get a `make clean` command for the build that hits all of the major targets, just to be sure on this and then we could eliminate one path from consideration.

**Comment by balhoff** [comment]
@kltm does the pipeline not start with a fresh ontology checkout each time? Would that be possible? I see another problem here that I didn't know about, which is that the ontology and version IRIs in the various products all point to `go.owl` rather than their particular variant.

**Comment by kltm** [comment]
@balhoff Sometimes it's a little hard to reason about the best way to do cleanups in Jenkins with docker images. Our particular setup has some quirks and it seems like in some cases it may not always be using a fresh checkout somehow, although we try to ensure it. (We also have some related permissions issues that we run into.) I agree that it could be some kind of sanitation causing this issue, but it is hard to pin down. Let's maybe explore a bot more here: https://github.com/geneontology/pipeline/issues/199

**Comment by murphyte** [comment]
Is it correct to consider the "2024-10-27" date as a bug?
We have some code that's extracting the date for version tracking, and happens to be extracting from the go-basic.oho file, hence our interest in that date. We can update our file as a workaround. I think we're getting the file through the InterProScan download, so we don't have other fields we could use as a better source of truth.

**Comment by kltm** [comment]
@murphyte Yes, I think it's correct to take is as a bug. If you need an external reference, there is https://current.geneontology.org/metadata/date.txt, as well as the listing at https://release.geneontology.org/ .

**Comment by kltm** [comment]
I have a possible fix in place (tracked at https://github.com/geneontology/pipeline/issues/199), but we'll have to watch the next release to see if it is effective.

**Comment by murphyte** [comment]
Great, thanks for the fix. We'll review with our next download.

**Comment by suzialeksander** [comment]
This looks to be resolved in 2025-03-16. Please open a new ticket if this reoccurs
